go/libkb/saltpack_sign.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2015 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package libkb
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/agl/ed25519"
+
+	keybase1 "github.com/keybase/client/go/protocol"
+	"github.com/keybase/client/go/saltpack"
+)
+
+func SaltPackSign(g *GlobalContext, source io.ReadCloser, sink io.WriteCloser, key NaclSigningKeyPair) error {
+	return saltpackSign(g, source, sink, key, saltpack.NewSignArmor62Stream)
+}
+
+func SaltPackSignDetached(g *GlobalContext, source io.ReadCloser, sink io.WriteCloser, key NaclSigningKeyPair) error {
+	return saltpackSign(g, source, sink, key, saltpack.NewSignDetachedArmor62Stream)
+}
+
+func saltpackSign(g *GlobalContext, source io.ReadCloser, sink io.WriteCloser, key NaclSigningKeyPair, streamer func(io.Writer, saltpack.SigningSecretKey) (io.WriteCloser, error)) error {
+	defer func() {
+		if err := source.Close(); err != nil {
+			g.Log.Warning("error closing source: %s", err)
+		}
+		if err := sink.Close(); err != nil {
+			g.Log.Warning("error closing sink: %s", err)
+		}
+	}()
+
+	stream, err := streamer(sink, saltSigner{key})
+	if err != nil {
+		return err
+	}
+
+	if _, err := io.Copy(stream, source); err != nil {
+		return err
+	}
+
+	if err = stream.Close(); err != nil {
+		return err
+	}
+
+	return nil
+}
```

Comment:
I'm not very familiar with our WriteCloser idioms, but this function feels pretty generic. Is there some not-saltpack-specific adapter we could use here?